### ISSUE TITLE
NPE Regression with Flutter 2.8+ and BannerAds

### ISSUE THREAD

**Issue by atrope** [googleads/googleads-mobile-flutter#495]
Hello, we have a major problem with latest Flutter 2.8+ and BannerAds in Android < 10.

Just like it was reported in #480 but closed, when we have the app run on Androids 7,8,9  and Have Banner ads in line with the
 
reply by @blasten here: https://github.com/googleads/googleads-mobile-flutter/issues/269#issuecomment-884493288
`  await PlatformViewsService.synchronizeToNativeViewHierarchy(false);
`
Every action that makes the app go to background (for example clicking the ad, or sharing some content) makes the app crash.

This is in our production app. 

Removing the line or Downgrading flutter seems to fix the issue but i consider this a major regression considering it was introduced by Flutter 2.8+ and it is now the recommended way of having inline ads on older devices..

Here is the example app, just run it on Old Android and click the button or the Example banner.

https://github.com/atrope/my_app

Forgot to add the LOG:

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.view.View.dispatchWindowVisibilityChanged(int)' on a null object reference
       at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
       at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
       at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
       at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1660)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1348)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6865)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:887)
       at android.view.Choreographer.doCallbacks(Choreographer.java:699)
       at android.view.Choreographer.doFrame(Choreographer.java:634)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:873)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6351)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:896)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:786)
```

**Comment by atrope** [comment]
Forgot to add the LOG:

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.view.View.dispatchWindowVisibilityChanged(int)' on a null object reference
       at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
       at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
       at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
       at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1660)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1348)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6865)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:887)
       at android.view.Choreographer.doCallbacks(Choreographer.java:699)
       at android.view.Choreographer.doFrame(Choreographer.java:634)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:873)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6351)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:896)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:786)
```

**Comment by maheshmnj** [comment]
Hi @atrope, Thanks for filing the issue. This seems to be the same issue as https://github.com/flutter/flutter/issues/96875, Please lets wait for the teams input on the issue filed in `flutter/flutter`. I will leave this issue open until we have an update on this.
This issue triage details can be found here https://github.com/flutter/flutter/issues/96875#issuecomment-1017232275

**Comment by doctek235** [comment]
Hello @maheshmnj!

>a major regression considering it was introduced by Flutter 2.8+

We're experiencing same problems since Flutter 2.8.1 so we have to fallback to version 2.5.3. 
But now it's even worse with 2.10 because it causes app crashes.

Is there any possible workaround exists to eliminate it temporary until https://github.com/flutter/flutter/issues/96875 will be resolved?

**Comment by maheshmnj** [comment]
> We're experiencing same problems since Flutter 2.8.1 so we have to fallback to version 2.5.3.
But now it's even worse with 2.10 because it causes app crashes.

@doctek235 Unfortunately, we will have to wait for the fix to land and we should eventually see an update on the flutter issue soon.

> We're experiencing same problems since Flutter 2.8.1 so we have to fallback to version 2.5.3.

Is there anything blocking you on stable 2.5.3?

**Comment by doctek235** [comment]
>Is there anything blocking you on stable 2.5.3?

@maheshmnj we affected by some minor bugs in 2.5.3 which seems fixed in 2.10 but nothing major.

I guess the major thing is expected performance improvement & stability of the app (in 2.10 compared to 2.5.3).

There is possible workaround proposed by @kolbasan mentioned in https://github.com/flutter/flutter/issues/96875#issuecomment-1024208495
>await PlatformViewsService.synchronizeToNativeViewHierarchy(false)

They using google ads package also so may be it can be applied as temporar fix.

**Comment by jjliu15** [comment]
This was resolved in master, per https://github.com/flutter/flutter/issues/96875#issuecomment-1112736710
